lib/ui/src/containers/nav.js

```diff
@@ -82,14 +82,14 @@ const createMenu = memoize(1)((api, shortcutKeys, isFullscreen, showPanel, showN
   {
     id: 'about',
     title: 'About your Storybook',
-    onClick: () => api.showSettingsPage('about'),
+    onClick: () => api.navigate('/settings/about'),
```

Comment:
@lonyele Doesn't this change fail to change the viewMode?

Reply:
No! I made the new api because I wasn't sure the place where assigning value to `viewMode`. However I confirmed that [parsePath](https://github.com/storybookjs/storybook/pull/7241/files#diff-18efc3f2c976feeab3887b2fd2a58f7eL50)  is the very first place where assigning value to `viewMode`

The reason why this bug happened is because `settings` viewMode was excluded from assigning value thus always return `undefined` which makes viewMode at the component level always [become a `story`](https://github.com/storybookjs/storybook/pull/7241/files#diff-0c70f9fbea9f0b5226ee80eb0cd454b4L59). At the end, url is `/settings/about` but viewMode is `story` thus showing both story preview component and settings page component.
```
const knownNonViewModesRegex = /(settings)/;
export const parsePath: (path?: string) => StoryData = memoize(1000)(
  (path: string | undefined | null) => {
    const result: StoryData = {
      viewMode: undefined,
      storyId: undefined,
    };

    if (path) {
      const [, viewMode, storyId] = path.match(splitPathRegex) || [undefined, undefined, undefined];
      if (viewMode && !viewMode.match(knownNonViewModesRegex)) {
        Object.assign(result, {
          viewMode,
          storyId,
        });
      }
    }
    return result;
  }
);
```